src/main/scala/gen/gmos/GmosNLongslitD.scala

```diff
@@ -64,17 +61,15 @@ sealed trait GmosNLongslitD[F[_]] {
    * effect evalutes `true`. Computes the science sequence as a stream of
    * "science/flat" "atoms" where the offset in Q and observing wavelength vary.
    *
-   * @param itc integration time calculator
    * @param reachedS2N a program that can be evaluated to determine whether the
    *                   desired s/n ratio has been reached
    *
    * @return a science sequence that termines when the desired s/n ratio is
    *         achieved
    */
   def science(
-    itc:        ImaginaryItc[F],
-    reachedS2N: F[Boolean]
-  ): Stream[F, Stream[F, Step.GmosN]]
+    reachedS2N: Int => F[Boolean]
+  ): Stream[F, Stream[F, (Step.GmosN, Double)]]
```

Comment:
The sequence now includes the step's contribution to reaching the S/N goal.  This needs a better type than a raw `Double` but I noticed the target S/N is a raw `Int`.